v2/event/event_marshal.go

```diff
@@ -150,7 +150,9 @@ func WriteJson(in *Event, writer io.Writer) error {
 			mediaType = strings.TrimSpace(strings.ToLower(contentType[0:i]))
 		}
 
-		isJson := mediaType == "" || mediaType == ApplicationJSON || mediaType == TextJSON
+		isJson := mediaType == "" ||
+			mediaType == ApplicationJSON || mediaType == TextJSON ||
+			mediaType == ApplicationCloudEventsJSON || mediaType == ApplicationCloudEventsBatchJSON
 
 		// If isJson and no encoding to base64, we don't need to perform additional steps
 		if isJson && !isBase64 {
```

Comment:
nit: suggesting for readability/future extension

```suggestion
jsonMediaTypes := map[string]bool{
    "":                                 true,
    ApplicationJSON:                   true,
    TextJSON:                          true,
    ApplicationCloudEventsJSON:       true,
    ApplicationCloudEventsBatchJSON:  true,
}

if jsonMediaTypes[mediaType] && !isBase64 {

```

Reply:
thanks for the neater go :bow: